---
title: "GWAS Analysis"
author: "LucianoRogerio"
date: "2022-04-26"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

```{r}
library(tidyverse)
library(BiocManager); library(compiler); library(sommer)
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install(version = "3.16")
source("http://www.zzlab.net/GAPIT/emma.txt")
source("https://zzlab.net/GAPIT/gapit_functions.txt")

Mrkrs <- readRDS(file = here::here("output", "GWASDoencasMkrs.rds")) %>% as.data.frame
Mrkrs$taxa <- rownames(Mrkrs)
Mrkrs <- Mrkrs %>% dplyr::select(taxa, everything())

MrkrsInf <- data.frame("SNP" = colnames(Mrkrs)[-1]) %>% 
  dplyr::mutate("Chromosome" = as.numeric(gsub("chr([0-9]+):([0-9]+)", "\\1", SNP)),
                "Position" = as.numeric(gsub("chr([0-9]+):([0-9]+)", "\\2", SNP)))

drgBlups <- read.table(file = here::here("output", "drgBlupsDoencas.csv"), header = T, sep = ",") %>% 
  dplyr::rename(taxa = clones)

myGAPIT <- GAPIT(
Y=drgBlups, 
GD=Mrkrs,
GM=MrkrsInf,
PCA.total=5,
ncpus = 3)

Model.selection=TRUE
model=c("GLM","MLM","CMLM","FarmCPU")

```

